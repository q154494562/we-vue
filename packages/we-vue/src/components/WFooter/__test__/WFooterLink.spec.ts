import { shallowMount } from '@vue/test-utils'
import FooterLink from '../WFooterLink'

describe('footer-link', () => {
  test('create', () => {
    const wrapper = shallowMount(FooterLink, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-footer-link')
    expect(wrapper.classes()).toContain('weui-footer__link')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('text', () => {
    const wrapper = shallowMount(FooterLink, {
      propsData: {
        text: 'test',
      },
    })

    expect(wrapper.text()).toBe('test')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('handle click', () => {
    const routeLinkSpy = jest.fn()
    const wrapper = shallowMount(FooterLink, {
      propsData: {},
      methods: {
        routeLink: routeLinkSpy,
      },
    })

    wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
    expect(routeLinkSpy).toHaveBeenCalled()
  })
})
