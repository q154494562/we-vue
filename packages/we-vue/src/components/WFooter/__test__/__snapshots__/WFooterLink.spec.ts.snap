// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer-link create 1`] = `<div class="weui-footer__link"></div>`;

exports[`footer-link text 1`] = `<div class="weui-footer__link">test</div>`;
